packages/react-components/react-tree/src/components/Tree/useTree.ts

```diff
@@ -12,13 +14,94 @@ import type { TreeProps, TreeState } from './Tree.types';
  * @param ref - reference to root HTMLElement of Tree
  */
 export const useTree_unstable = (props: TreeProps, ref: React.Ref<HTMLElement>): TreeState => {
+  const {
+    isSubtree,
+    level: parentLevel,
+    openTrees: rootOpenTrees,
+    requestOpenChange: rootRequestOpenChange,
+    treeRef: parentTreeRef,
+  } = useTreeContext_unstable();
+  warnIfNecessary(props, isSubtree);
+  const { openSubtrees: stateOpenTrees, defaultOpenSubtrees: defaultOpenTrees, onOpenChange, ...rest } = props;
+  const [localOpenTrees, setOpenTrees] = useControllableState({
+    state: React.useMemo(() => normalizeOpenTreesOrUndefined(stateOpenTrees), [stateOpenTrees]),
+    defaultState: () => normalizeOpenTrees(defaultOpenTrees),
+    initialState: [],
+  });
+  const arrowNavigationProps = useArrowNavigationGroup({
+    tabbable: true,
+    axis: 'vertical',
+  });
+  const localRequestOpenChange = React.useCallback(
+    (data: TreeOpenChangeData) => {
+      onOpenChange?.(data.event, data);
+      if (!data.event.isDefaultPrevented()) {
+        setOpenTrees(curr => updateOpenTrees(data, curr));
+      }
+    },
+    [onOpenChange, setOpenTrees],
+  );
+  const openTrees = isSubtree ? rootOpenTrees : localOpenTrees;
+  const requestOpenChange = isSubtree ? rootRequestOpenChange : localRequestOpenChange;
+  const isOpen = React.useMemo(() => !isSubtree || props.id === undefined || openTrees.includes(props.id), [
+    props.id,
+    openTrees,
```

Comment:
this memo should be quite redundant since each render will most likely create a new reference of `openTrees`

Reply:
You're probably right, I don't see any other case for re-rendering but the change of `openTrees`, I'll investigate it